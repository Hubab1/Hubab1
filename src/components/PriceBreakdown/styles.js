import {
    css
} from 'emotion';

export const existingItemRow = css`
    border-top: 1px solid #EEEEEE;
    padding: 10px 0;
    display: flex;
    text-align: left;
    justify-content: space-between;
`;

export const priceBreakdownContainer = css`
    .MuiExpansionPanelSummary-root {
        padding: 0;
        border-top: 1px solid #EEEEEE;
    }
`;
