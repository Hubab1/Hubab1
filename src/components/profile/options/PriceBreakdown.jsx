import React, { useState, useEffect } from "react";
import PropTypes from 'prop-types';

import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Box from '@material-ui/core/Box';
import {makeStyles} from '@material-ui/core/styles';
import styled from '@emotion/styled';

import API  from 'app/api';

import { P, CardRowTotal } from 'assets/styles';
import {
    anchor,
    contentContainer, existingItemRow,
    priceBreakdownContainer,
    paperRoot,
    renterProfileListItemContainer,
} from './styles';
import Tip from 'components/common/Tip';


const useStyles = makeStyles(theme => ({
    root: {
        display: 'block',
        padding: 0,
    },
}));

const PriceBreakdownContainer = styled.div`
    margin-bottom: 20px;
`

function PriceBreakdown (props) {
    const classes = useStyles();
    const [priceBreakdown, setPriceBreakdown] = useState({});
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        const body = {
            application: props.application.id,
            rental_options: props.selectedOptions,
            unit_id: props.unitId,
        };
        API.getCurrentFlatQuote(body).then((result) => {
            setPriceBreakdown(result);
            setIsLoading(false);
        });
    }, [props.application, props.selectedOptions, props.unitId]);

    const getCurrentCategoryInfo = () => {
        // Count the number of rental options selected
        const categoryCount = Object.values(props.selectedOptions).reduce((a, b) => (a + b), 0);

        let categoryMonthlyPrice = priceBreakdown.items_breakdown[props.category];

        let categoryInfo = 'Based on your selected rental options';
        if (categoryMonthlyPrice) {
            categoryInfo = `${categoryMonthlyPrice}/mo for ${categoryCount} ${props.categoryHelperText}`
        }

        return (<>{categoryInfo}</>)
    };

    return (
        <PriceBreakdownContainer>
            {!isLoading && (
                <div className={renterProfileListItemContainer}>
                    <div id={props.name} className={anchor}/>
                    <div className={paperRoot}>
                        <div className={contentContainer}>
                            <div>
                                <Box>
                                    <Tip
                                        text={
                                            <P>
                                                <b>{priceBreakdown.total} Total Rent</b><br/>
                                                {getCurrentCategoryInfo()}
                                            </P>
                                        }
                                    />
                                </Box>
                            </div>
                        </div>
                        <div className={priceBreakdownContainer}>
                            <ExpansionPanel elevation={0} defaultExpanded={true}>
                                <ExpansionPanelSummary
                                    expandIcon={<ExpandMoreIcon />}
                                >
                                    Rent breakdown
                                </ExpansionPanelSummary>
                                <ExpansionPanelDetails classes={{root:classes.root}}>
                                    <div className={existingItemRow}>
                                        Base Rent<span className={"pull-right"}>{priceBreakdown.base_rent}</span>
                                    </div>
                                    {
                                        Object.keys(priceBreakdown.items_breakdown).map(function(key) {
                                            return (
                                                <div key={key} className={existingItemRow}>
                                                    {key}<span className={"pull-right"}>{priceBreakdown.items_breakdown[key]}</span>
                                                </div>)
                                        })
                                    }
                                    <CardRowTotal>
                                        <P bold>Total</P>
                                        <div>
                                            <P bold>{priceBreakdown.total}</P>
                                        </div>
                                    </CardRowTotal>
                                </ExpansionPanelDetails>
                            </ExpansionPanel>
                        </div>
                    </div>
                </div>
            )}
        </PriceBreakdownContainer>
    );
}

PriceBreakdown.propTypes = {
    application: PropTypes.object,
    selectedOptions: PropTypes.object,
    unitId: PropTypes.number,
    category: PropTypes.string,
    categoryHelperText: PropTypes.string,
}

export default PriceBreakdown;