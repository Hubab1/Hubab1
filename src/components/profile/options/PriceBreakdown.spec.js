import {mount} from 'enzyme';
import {act} from 'react-dom/test-utils';
import React from 'react';
import API from 'app/api';
import PriceBreakdown from 'components/profile/options/PriceBreakdown';

const buildProps = () => {
    return {
        application: { id: 123, },
        selectedOptions: {},
        unitId: 12,
        baseRent: 1200,
        category: 'storage',
        categoryHelperText: 'storage space',
    }
};

describe('Price Breackdown section', () => {
    it('display correct info', async () => {
        const props = buildProps();
        API.getCurrentFlatQuote = jest.fn().mockReturnValue(Promise.resolve({
            total: '$2,020',
            items_breakdown: {
                storage: '$10',
                pets: '$10'
            },
            base_rent: '$2,000'
        }));
        let wrapper = mount(<PriceBreakdown {...props} />);
        await act(async () => {
            await Promise.resolve(wrapper);
            wrapper.update();
        });
        expect(wrapper.debug()).toMatchSnapshot();
    })
})