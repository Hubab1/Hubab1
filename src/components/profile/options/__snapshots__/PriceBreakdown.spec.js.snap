// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Breackdown section display correct info 1`] = `
"<PriceBreakdown application={{...}} selectedOptions={{...}} unitId={12} category=\\"storage\\" categoryHelperText=\\"storage space\\">
  <Styled(div)>
    <div className=\\"css-1gbujtk\\">
      <div className=\\"css-101ez95\\">
        <div id={[undefined]} className=\\"css-1h1izsk\\" />
        <div className=\\"css-coqf6u\\">
          <div className=\\"css-hgfyev\\">
            <div>
              <Styled(MuiBox)>
                <div className=\\"MuiBox-root MuiBox-root-3\\">
                  <Tip text={{...}}>
                    <WithStyles(ForwardRef(Grid)) container={true} spacing={1} alignItems=\\"center\\">
                      <ForwardRef(Grid) classes={{...}} container={true} spacing={1} alignItems=\\"center\\">
                        <div className=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center\\">
                          <WithStyles(ForwardRef(Grid)) item={true}>
                            <ForwardRef(Grid) classes={{...}} item={true}>
                              <div className=\\"MuiGrid-root MuiGrid-item\\">
                                <Styled(img) alt=\\"light bulb\\" src=\\"lightbulb.png\\">
                                  <img alt=\\"light bulb\\" src=\\"lightbulb.png\\" className=\\"css-9ojriw\\" />
                                </Styled(img)>
                              </div>
                            </ForwardRef(Grid)>
                          </WithStyles(ForwardRef(Grid))>
                          <WithStyles(ForwardRef(Grid)) item={true} xs={true}>
                            <ForwardRef(Grid) classes={{...}} item={true} xs={true}>
                              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\">
                                <WithStyles(ForwardRef(Grid)) item={true} xs={true}>
                                  <ForwardRef(Grid) classes={{...}} item={true} xs={true}>
                                    <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\">
                                      <Styled(b) fontSize={16}>
                                        <b fontSize={16} className=\\"css-1xtqd4a\\" />
                                      </Styled(b)>
                                    </div>
                                  </ForwardRef(Grid)>
                                </WithStyles(ForwardRef(Grid))>
                                <WithStyles(ForwardRef(Grid)) item={true} xs={true}>
                                  <ForwardRef(Grid) classes={{...}} item={true} xs={true}>
                                    <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\">
                                      <Styled(p)>
                                        <p className=\\"css-3m08t9\\">
                                          <b>
                                            $2,020
                                             Total Rent
                                          </b>
                                          <br />
                                          $10/mo for 0 storage space
                                        </p>
                                      </Styled(p)>
                                    </div>
                                  </ForwardRef(Grid)>
                                </WithStyles(ForwardRef(Grid))>
                              </div>
                            </ForwardRef(Grid)>
                          </WithStyles(ForwardRef(Grid))>
                        </div>
                      </ForwardRef(Grid)>
                    </WithStyles(ForwardRef(Grid))>
                  </Tip>
                </div>
              </Styled(MuiBox)>
            </div>
          </div>
          <div className=\\"css-1bh0d37\\">
            <WithStyles(ForwardRef(ExpansionPanel)) elevation={0} defaultExpanded={false}>
              <ForwardRef(ExpansionPanel) classes={{...}} elevation={0} defaultExpanded={false}>
                <WithStyles(ForwardRef(Paper)) className=\\"MuiExpansionPanel-root MuiExpansionPanel-rounded\\" square={false} elevation={0}>
                  <ForwardRef(Paper) classes={{...}} className=\\"MuiExpansionPanel-root MuiExpansionPanel-rounded\\" square={false} elevation={0}>
                    <div className=\\"MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation0 MuiPaper-rounded\\">
                      <WithStyles(ForwardRef(ExpansionPanelSummary)) expandIcon={{...}}>
                        <ForwardRef(ExpansionPanelSummary) classes={{...}} expandIcon={{...}}>
                          <WithStyles(ForwardRef(ButtonBase)) focusRipple={false} disableRipple={true} disabled={false} component=\\"div\\" aria-expanded={false} className=\\"MuiExpansionPanelSummary-root\\" onFocusVisible={[Function: handleFocusVisible]} onBlur={[Function: handleBlur]} onClick={[Function: handleChange]}>
                            <ForwardRef(ButtonBase) classes={{...}} focusRipple={false} disableRipple={true} disabled={false} component=\\"div\\" aria-expanded={false} className=\\"MuiExpansionPanelSummary-root\\" onFocusVisible={[Function: handleFocusVisible]} onBlur={[Function: handleBlur]} onClick={[Function: handleChange]}>
                              <div className=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} aria-expanded={false}>
                                <div className=\\"MuiExpansionPanelSummary-content\\">
                                  Rent Breakdown
                                </div>
                                <WithStyles(ForwardRef(IconButton)) className=\\"MuiExpansionPanelSummary-expandIcon\\" edge=\\"end\\" component=\\"div\\" tabIndex={{...}} role={{...}} aria-hidden={true}>
                                  <ForwardRef(IconButton) classes={{...}} className=\\"MuiExpansionPanelSummary-expandIcon\\" edge=\\"end\\" component=\\"div\\" tabIndex={{...}} role={{...}} aria-hidden={true}>
                                    <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" centerRipple={true} focusRipple={true} disabled={false} component=\\"div\\" tabIndex={{...}} role={{...}} aria-hidden={true}>
                                      <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" centerRipple={true} focusRipple={true} disabled={false} component=\\"div\\" tabIndex={{...}} role={{...}} aria-hidden={true}>
                                        <div className=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" onBlur={[Function]} onClick={[undefined]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={{...}} role={{...}} aria-disabled={false} aria-hidden={true}>
                                          <span className=\\"MuiIconButton-label\\">
                                            <ForwardRef>
                                              <WithStyles(ForwardRef(SvgIcon))>
                                                <ForwardRef(SvgIcon) classes={{...}}>
                                                  <svg className=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role=\\"presentation\\">
                                                    <path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\" />
                                                    <path fill=\\"none\\" d=\\"M0 0h24v24H0z\\" />
                                                  </svg>
                                                </ForwardRef(SvgIcon)>
                                              </WithStyles(ForwardRef(SvgIcon))>
                                            </ForwardRef>
                                          </span>
                                          <NoSsr />
                                        </div>
                                      </ForwardRef(ButtonBase)>
                                    </WithStyles(ForwardRef(ButtonBase))>
                                  </ForwardRef(IconButton)>
                                </WithStyles(ForwardRef(IconButton))>
                              </div>
                            </ForwardRef(ButtonBase)>
                          </WithStyles(ForwardRef(ButtonBase))>
                        </ForwardRef(ExpansionPanelSummary)>
                      </WithStyles(ForwardRef(ExpansionPanelSummary))>
                      <WithStyles(ForwardRef(Collapse)) in={false} timeout=\\"auto\\">
                        <ForwardRef(Collapse) classes={{...}} in={false} timeout=\\"auto\\">
                          <Transition in={false} onEnter={[Function: handleEnter]} onEntered={[Function: handleEntered]} onEntering={[Function: handleEntering]} onExit={[Function: handleExit]} onExiting={[Function: handleExiting]} addEndListener={[Function: addEndListener]} timeout={{...}} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onExited={[Function: noop]}>
                            <div className=\\"MuiCollapse-container MuiCollapse-hidden\\" style={{...}}>
                              <div className=\\"MuiCollapse-wrapper\\">
                                <div className=\\"MuiCollapse-wrapperInner\\">
                                  <div aria-labelledby={[undefined]} id={[undefined]} role=\\"region\\">
                                    <WithStyles(ForwardRef(ExpansionPanelDetails)) classes={{...}}>
                                      <ForwardRef(ExpansionPanelDetails) classes={{...}}>
                                        <div className=\\"MuiExpansionPanelDetails-root makeStyles-root-1\\">
                                          <div className=\\"css-olqeuu\\">
                                            Base Rent
                                            <span className=\\"pull-right\\">
                                              $2,000
                                            </span>
                                          </div>
                                          <div>
                                            <div className=\\"css-olqeuu\\">
                                              storage
                                              <span className=\\"pull-right\\">
                                                $10
                                              </span>
                                            </div>
                                          </div>
                                          <div>
                                            <div className=\\"css-olqeuu\\">
                                              pets
                                              <span className=\\"pull-right\\">
                                                $10
                                              </span>
                                            </div>
                                          </div>
                                          <Styled(div)>
                                            <div className=\\"css-zgrhnt\\">
                                              <Styled(p) bold={true}>
                                                <p className=\\"css-1bj6jbc\\">
                                                  Total
                                                </p>
                                              </Styled(p)>
                                              <div>
                                                <Styled(p) bold={true}>
                                                  <p className=\\"css-1bj6jbc\\">
                                                    $2,020
                                                  </p>
                                                </Styled(p)>
                                              </div>
                                            </div>
                                          </Styled(div)>
                                        </div>
                                      </ForwardRef(ExpansionPanelDetails)>
                                    </WithStyles(ForwardRef(ExpansionPanelDetails))>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </Transition>
                        </ForwardRef(Collapse)>
                      </WithStyles(ForwardRef(Collapse))>
                    </div>
                  </ForwardRef(Paper)>
                </WithStyles(ForwardRef(Paper))>
              </ForwardRef(ExpansionPanel)>
            </WithStyles(ForwardRef(ExpansionPanel))>
          </div>
        </div>
      </div>
    </div>
  </Styled(div)>
</PriceBreakdown>"
`;
